### PR TITLE
sccheduler, server: persist scheudler list to etcd

### PR DESCRIPTION
**Pull request by Connor1996**
This is the recommitment for #769, and fix compatible problem

### CODE REVIEW

File: server/config.go [hunk 1]
```diff
@@ -308,7 +308,7 @@ type ScheduleConfig struct {
 	// ReplicaScheduleLimit is the max coexist replica schedules.
 	ReplicaScheduleLimit uint64 `toml:"replica-schedule-limit,omitempty" json:"replica-schedule-limit"`
 	// Schedulers support for loding customized schedulers
-	Schedulers SchedulerConfigs `toml:"schedulers,omitempty" json:"schedulers"`
+	Schedulers SchedulerConfigs `toml:"schedulers,omitempty" json:"schedulers-v2"` // json v2 is the for sake of compatible upgrade
```
Comment (disksing) [inline server/config.go]: is the for -> is for.

Comment (Connor1996) [reply server/config.go]: "the" is necessary here, it's idiomatic usage

Comment (Connor1996) [reply server/config.go]: "the" is necessary, but it should after "for", I mistake here...

Comment (disksing) [reply server/config.go]: got it.

Comment (siddontang) [reply server/config.go]: why `-v2` here?

Comment (Connor1996) [reply server/config.go]: schedulerConfigs struct in old version is different from new version, so when reload persisted old version json string, new version can not unmarshal correctly. Thus change the name of this field, new version will ignore the old version info which in field "schedulers" and store new info in "schedulers-v2".

**Pull request merged**